pkgs/os-specific/linux/zfs/generic.nix

```diff
@@ -235,7 +241,7 @@ let
       mainProgram = "zfs";
       # If your Linux kernel version is not yet supported by zfs, try zfs_unstable.
       # On NixOS set the option `boot.zfs.package = pkgs.zfs_unstable`.
-      broken = buildKernel && (kernelCompatible != null) && !kernelCompatible;
+      broken = buildKernel && (kernelCompatible kernel != null) && !(kernelCompatible kernel);
```

Comment:
I think the original `kernelCompatible != null` check should be left as-is since `kernelCompatible` is an optional attribute.